AdvLoggerPkg/Include/Guid/AdvancedLoggerPreDxeLogs.h

```diff
@@ -0,0 +1,33 @@
+/** @file
+  This file defines GUIDs and data structure for producing HOB entry describing
+  the location of pre-DXE logs. For platforms which use a Pre-DXE AdvancedLoggerLib
+  implementation, this HOB is not necessary.
+
+  Copyright (C) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef ADVANCED_LOGGER_PRE_DXE_LOGS_H_
+#define ADVANCED_LOGGER_PRE_DXE_LOGS_H_
+
+#pragma pack(1)
+// The region specified by this HOB entry should contain raw log output.
+// The Advanced Logger library will serialize the contents of the buffer
+// when adding it to the advanced logger output.
+typedef struct _ADVANCED_LOGGER_PRE_DXE_LOGS_HOB {
+  UINT32    Signature;
+  UINT64    BaseAddress;
+  UINT64    LengthInBytes;
+} ADVANCED_LOGGER_PRE_DXE_LOGS_HOB;

```

Comment:
Should the HOB be versioned, since these may come from different compilation environments?

Reply:
I don't *think* this hob would ever need to change. If it does, then I think we can change the signature or add a new v2 HOB definition.